File: TP3_CPP/Codingan/CPU.cpp
#pragma once
#include "Internal.cpp"
#include <iostream>
#include <string>

using namespace std;

class CPU : public Internal
{
private:
    int JumlahCore;
    float KecepatanGHZ;

public:
    CPU()
    {
    }

    CPU(int JumlahCore, float KecepatanGHZ, string Nama, string Spesifikasi, string MerkINTERNAL) : Internal(Nama, Spesifikasi, MerkINTERNAL)
    {
        this->JumlahCore = JumlahCore;
        this->KecepatanGHZ = KecepatanGHZ;
    }

    // Setter
    void setJumlahCore(int JumlahCore)
    {
        this->JumlahCore = JumlahCore;
    }

    void setKecepatanGHZ(float KecepatanGHZ)
    {
        this->KecepatanGHZ = KecepatanGHZ;
    }

    // Getter
    int getJumlahCore()
    {
        return JumlahCore;
    }

    float getKecepatanGHZ()
    {
        return KecepatanGHZ;
    }

    ~CPU()
    {
    }
};

File: TP3_Python/Codingan/CPU.py
from Internal import Internal

class CPU(Internal):
    def __init__(self, jumlah_core=0, kecepatan_ghz=0.0, nama="", spesifikasi="", merk_internal=""):
        super().__init__(nama, spesifikasi, merk_internal)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz

    def set_jumlah_core(self, jumlah_core):
        self.jumlah_core = jumlah_core

    def set_kecepatan_ghz(self, kecepatan_ghz):
        self.kecepatan_ghz = kecepatan_ghz

    def get_jumlah_core(self):
        return self.jumlah_core

    def get_kecepatan_ghz(self):
        return self.kecepatan_ghz